addon/components/async-button.js

```diff
@@ -38,9 +38,18 @@ const ButtonComponent = Component.extend(positionalParamsMixin, {
       set(this, 'promise', promise);
     };
 
-    let actionArguments = ['action', callbackHandler, ...params];
+    if (typeof this.attrs.action === 'function') {
+      let promise = this.attrs.action();
```

Comment:
:+1: All great points -- I'll make these changes and get this back for another review.